modThread/modThread.py

```diff
@@ -4,280 +4,404 @@
+            title="ModThread Primary Category Settings",
+            description="Primary Category configuration for modThread.",
+            color=discord.Color.blue(),
+        )
 
-    @commands.guild_only()
-    @commands.command()
-    @checks.admin_or_permissions(manage_guild=True)
-    async def getNumbersCategory(self, ctx):
-        """Gets the category currently assigned as the numbers category"""
-        try:
-            await ctx.send(
-                "Numbers Thread category set to: {0}".format(
-                    (await self._numbers_category(ctx)).mention
+        if category is not None:
+            if category in ['delete', 'rm', 'del', 'clear', 'unset']:
+                category = category.lower()
+                await self._set_primary_category(ctx.guild, None)
+                settings_embed.add_field(
+                    name="Primary Category Removed",
+                    value="Not Set",
+                    inline=False
+                )
+            else:
+                set_category = await self._set_primary_category(
+                    ctx.guild,
+                    category
                 )
+                settings_embed.add_field(
+                    name="Category Set",
+                    value=f"Category set to {set_category.jump_url}",
+                    inline=False
+                )
+        else:
+            primary_category = await self._get_primary_category(ctx.guild)
+            settings_embed.add_field(
+                name="Current Category",
+                value=f"Category set to {primary_category.jump_url}",
+                inline=False
             )
-        except:
-            await ctx.send(":x: Numbers Thread category not set")
-
-    @commands.guild_only()
-    @commands.command()
-    @checks.admin_or_permissions(manage_guild=True)
-    async def unsetNumbersCategory(self, ctx):
-        """Unsets the numbers category. Thread will not be moved if no numbers category is set"""
-        await self._save_numbers_category(ctx, None)
-        await ctx.send("Done")
 
-    ### End Numbers Category
-
-    ### Mods Category
-    @commands.guild_only()
-    @commands.command()
-    @checks.admin_or_permissions(manage_guild=True)
-    async def setModsRole(self, ctx, mods_role: discord.Role):
-        await self._save_mods_role(ctx, mods_role.id)
-        await ctx.send("Done")
-
-    @commands.guild_only()
-    @commands.command()
-    @checks.admin_or_permissions(manage_guild=True)
-    async def setModsCategory(self, ctx, mods_category: discord.CategoryChannel):
-        """Sets the category where all transaction messages will be posted"""
-        await self._save_mods_category(ctx, mods_category.id)
-        await ctx.send("Done")
+        await ctx.send(embed=settings_embed)
+
+    @modthread.command(name="role")
+    async def management_role(
+        self,
+        ctx: commands.Context,
+        role: discord.Role | str | None
```

Comment:
Same as category typing. Remove str

Reply:
Same as above. Needed for the overload in arguments